cpp/src/slice2cs/Gen.cpp

```diff
@@ -1726,6 +1711,10 @@ Slice::Gen::TypesVisitor::visitClassDefEnd(const ClassDefPtr& p)
         for (const auto& q : dataMembers)
         {
             const string memberName = fixId(q->name(), DotNet::ICloneable);
+            if (!q->optional() && isNonNullableReferenceType(q->type()))
+            {
+                _out << nl << "global::System.ArgumentNullException.ThrowIfNull(" << memberName << ");";
```

Comment:
We should pass the member name to get a better error message, like in:
```suggestion
                _out << nl << "global::System.ArgumentNullException.ThrowIfNull(" << memberName << ", nameof(" << memberName << "));";
```

Reply:
No, this would be incorrect as it's done automatically already.
See Remarks of https://learn.microsoft.com/en-us/dotnet/api/system.argumentnullexception.throwifnull?view=net-8.0.